src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -3360,7 +3369,15 @@ protected void checkThumbnail(InputFile inputFile, RendererConfiguration rendere
 
 			media.generateThumbnail(inputFile, getFormat(), getType(), seekPosition, isResume(), renderer);
 			if (!isResume() && media.getThumb() != null && configurationSpecificToRenderer.getUseCache() && inputFile.getFile() != null) {
-				MediaTableThumbnails.setThumbnail(media.getThumb(), inputFile.getFile().getAbsolutePath(), -1);
+				Connection connection = null;
+				try {
+					connection = MediaDatabase.getConnectionIfAvailable();
+					if (connection != null) {
+						MediaTableThumbnails.setThumbnail(connection, media.getThumb(), inputFile.getFile().getAbsolutePath(), -1);
+					}
+				} finally {
+					MediaDatabase.close(connection);
+				}
```

Comment:
I wonder if we can have the best of both worlds by _allowing_ an incoming connection, but if one doesn't exist, the method itself (in this case `MediaTableThumbnails.setThumbnail`) makes a new one?
That way if we have a chain of db operations to do, we can use the same connection, but in cases like this one we don't have to create one in the caller for just one operation.
Is that too messy logically? I'm interested in your opinion

Reply:
That's what I have done before, and then I removed it.
When I removed it, I had not yet implemented all the stuff (db state, ...).

So it was reopening the db on connection creation call, even if we stopped it.

By the way, some functions need to access the db even if its state is not OPENED (like table checks).

But, I'm now able to recreate some function without connection if you want.
I'll do it for `MediaTableThumbnails.setThumbnail`